pyo3-macros-backend/src/pyclass.rs

```diff
@@ -413,54 +414,66 @@ fn impl_enum(
 ) -> syn::Result<TokenStream> {
     let enum_name = &enum_.ident;
     let doc = utils::get_doc(&enum_.attrs, None);
-    let enum_cls = impl_enum_class(enum_name, &attrs, doc, methods_type)?;
-
-    let variant_consts = variants
-        .iter()
-        .map(|v| impl_const(enum_name, v.ident))
-        .collect::<syn::Result<Vec<_>>>()?;
+    let enum_cls = impl_enum_class(enum_name, &attrs, &variants, doc, methods_type)?;
 
     Ok(quote! {
-
         #enum_cls
-
-        #[pymethods]
-        impl #enum_name {
-            #(
-                #[allow(non_upper_case_globals)]
-                #variant_consts
-            )*
-        }
-    })
-}
-
-fn impl_const(enum_: &syn::Ident, cls: &syn::Ident) -> syn::Result<TokenStream> {
-    Ok(quote! {
-        #[classattr]
-        const #cls: #enum_ = #enum_::#cls;
     })
 }
 
 fn impl_enum_class(
     cls: &syn::Ident,
     attr: &PyClassArgs,
+    variants: &Vec<VariantPyO3>,
     doc: PythonDoc,
     methods_type: PyClassMethodsType,
 ) -> syn::Result<TokenStream> {
     let pytypeinfo = impl_pytypeinfo(cls, attr, None);
     let pyclass_impls = PyClassImplsBuilder::new(cls, attr, methods_type)
         .doc(doc)
         .impl_all();
+    let descriptors = unit_variants_as_descriptors(cls, variants.iter().map(|v| v.ident));
 
     Ok(quote! {
 
         #pytypeinfo
 
         #pyclass_impls
 
+        #descriptors
+
     })
 }
 
+fn unit_variants_as_descriptors<'a>(
+    cls: &'a syn::Ident,
+    variant_names: impl IntoIterator<Item = &'a syn::Ident>,
+) -> TokenStream {
+    let cls_type = syn::parse_quote!(#cls);
+    let variant_to_attribute = |ident: &syn::Ident| ConstSpec {
```

Comment:
This is dirty. Have no idea how to change that though.

Reply:
What's your dislike about it? It seems ok to me go through the existing machinery.

Reply:
I... can't remember 😅 I guess I wanted some constructor function.